packages/smooth_app/lib/cards/category_cards/svg_cache.dart

```diff
@@ -80,14 +81,56 @@ class SvgCache extends AbstractCache {
       'nova-group-3.svg' => localizations.nova_group_3,
       'nova-group-4.svg' => localizations.nova_group_4,
       'nova-group-unknown.svg' => localizations.nova_group_unknown,
-      'nutriscore-a.svg' => localizations.nutriscore_a,
-      'nutriscore-b.svg' => localizations.nutriscore_b,
-      'nutriscore-c.svg' => localizations.nutriscore_c,
-      'nutriscore-d.svg' => localizations.nutriscore_d,
-      'nutriscore-e.svg' => localizations.nutriscore_e,
-      'nutriscore-unknown.svg' => localizations.nutriscore_unknown,
-      'nutriscore-not-applicable.svg' => localizations.ecoscore_not_applicable,
+      String _ when fileName.startsWith('nutriscore-') =>
+        _extractNutriScore(localizations, fileName),
       _ => null,
     };
   }
+
+  static String _extractNutriScore(
+    AppLocalizations localizations,
+    String fileName,
+  ) {
+    // Old NutriScore
+    if (fileName == 'nutriscore-a.svg') {
+      return localizations.nutriscore_a;
+    } else if (fileName == 'nutriscore-b.svg') {
```

Comment:
I'm sure a `switch` syntax would be much easier to read.

Reply:
Unfortunately, with a `return switch` statement, it's mandatory to have an `else` case (didn't know that before trying after your suggestion).
I can "switch" to the old switch (with `cases`), but in terms of lines, it will be similar